### PR TITLE
Lets you add or remove modules from equipped armor

### PR DESCRIPTION
**Pull request by QualityVan**
## About The Pull Request
As title.

## Why It's Good For The Game
It's always a little annoying to do inventory management at round start with modular armor since all the parts vend separately.

## Changelog
:cl:
add: You can attach or remove modules from modular armor while you're still wearing it
/:cl:

### CODE REVIEW

**Review by BraveMole — CHANGES_REQUESTED** [review]
I don't think this is a proper way of doing it.

File: code/modules/modular_armor/modular.dm [hunk 1]
```diff
@@ -172,20 +173,32 @@
...
-		return FALSE
-
 	if(!do_after(user, equip_delay, TRUE, user, BUSY_ICON_GENERIC))
 		return FALSE
 
+///Temporarily unequips and then requips the armor during the attach process to keep armor datums/actions up to date if needed
+/obj/item/clothing/suit/modular/proc/do_attach(mob/living/user, obj/item/armor_module/module)
+	var/needs_requip = user.get_item_by_slot(SLOT_WEAR_SUIT) == src
+	if(needs_requip)
+		unequipped(user, SLOT_WEAR_SUIT)
```
Comment (BraveMole) [inline code/modules/modular_armor/modular.dm:183]: this drops your weapon?

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/modules/modular_armor/modular.dm (hunk 1 above)
Comment (QualityVan) [reply code/modules/modular_armor/modular.dm:183]: Why would it drop your weapon? It's a call on the armor itself. The armor stays in place, it just momentarily pretends it's unequipped in order to clear out actions and armor values, and then reapply it.

**Comment by QualityVan** [comment]
Should do this for helmets too, actually.

File: code/modules/modular_armor/modular.dm (hunk 1 above)
Comment (QualityVan) [reply code/modules/modular_armor/modular.dm:183]: Oh wait you mean your suit storage uhhhh that's a good question

Comment (QualityVan) [reply code/modules/modular_armor/modular.dm:183]: Okay it doesn't.

**Comment by QualityVan** [comment]
> 
> 
> I don't think this is a proper way of doing it.

Better to just reimplement the action/slowdown/armor removal equipped/unequipped does directly, you think?

**Review by BraveMole — APPROVED** [review]
Dis lazy but okay

File: code/modules/modular_armor/modular.dm [hunk 2]
```diff
@@ -172,20 +173,32 @@
...
 
+///Temporarily unequips and then requips the armor during the attach process to keep armor datums/actions up to date if needed
+/obj/item/clothing/suit/modular/proc/do_attach(mob/living/user, obj/item/armor_module/module)
+	var/needs_reequip = user.get_item_by_slot(SLOT_WEAR_SUIT) == src
+	if(needs_reequip)
+		unequipped(user, SLOT_WEAR_SUIT)
+	module.do_attach(user, src)
+	if(needs_reequip)
+		equipped(user, SLOT_WEAR_SUIT)
+
```
Comment (TiviPlus) [inline code/modules/modular_armor/modular.dm:187]: this does not actually use any extra utility of the suit this should be on the actual module
Additionally you named two procs the same leading to confusion but see above

File: code/modules/modular_armor/modular.dm [hunk 3]
```diff
@@ -172,20 +173,32 @@
...
 		return FALSE
 
+///Temporarily unequips and then requips the armor during the detach process to keep armor datums/actions up to date if needed
+/obj/item/clothing/suit/modular/proc/do_detach(mob/living/user, obj/item/armor_module/module)
+	var/needs_reequip = user.get_item_by_slot(SLOT_WEAR_SUIT) == src
+	if(needs_reequip)
+		unequipped(user, SLOT_WEAR_SUIT)
+	module.do_detach(user, src)
+	if(needs_reequip)
+		equipped(user, SLOT_WEAR_SUIT)
```
Comment (TiviPlus) [inline code/modules/modular_armor/modular.dm:201]: ditto

File: code/modules/modular_armor/modular.dm [hunk 4]
```diff
@@ -586,13 +582,31 @@
 		return FALSE
 	update_overlays()
 
+///Temporarily unequips and then requips the helmet during the module attach process to keep armor datums/actions up to date if needed
+/obj/item/clothing/head/modular/proc/do_attach(mob/living/user, obj/item/helmet_module/module)
+	var/needs_reequip = user.get_item_by_slot(SLOT_HEAD) == src
+	if(needs_reequip)
+		unequipped(user, SLOT_HEAD)
+	module.do_attach(user, src)
+	if(needs_reequip)
```
Comment (TiviPlus) [inline code/modules/modular_armor/modular.dm:591]: same for helmet as for suit

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Pull request closed without merging**